src/signature.c

```diff
@@ -871,6 +871,15 @@ gboolean cms_verify_file(const gchar *filename, GBytes *sig, gsize limit, X509_S
 	}
 	content = g_mapped_file_get_bytes(file);
 
+	if (limit > (goffset)G_MAXSIZE) {
```

Comment:
G_MAXSIZE is 4GB on 32 bit systems, did you mean G_MAXSSIZE?
mmapping even 2GB on 32 bit systems is unlikely to work due to limited free address space.

Reply:
I think G_MAXSIZE is right, the commit's explanation is incorrect, should be 4GB instead of 2GB. It actually aims to catch `limit` values exceeding the (potentially) mmappable size. However, one would probably run into memory issues also for smaller bundles.

Reply:
Now also got the point that casting G_MAXSIZE to goffset might not be such a good idea on 64bit systems... I've used (guint64) casts for both `limit` and G_MAXSIZE now.